### PR TITLE
[React-core] Fix caching issue with useTotalCirculatingSupply

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved
The code below doesn't work beecause currently they all share the same cacheKey, so as the result all of them will return the cached data from the first `useTotal...` (tokenId #0)

This fix adds the `tokenId` as the extra cacheKey
```
const { contract: nftContract } = useContract(
  "0x8a41F6......9829659526"
);
const { data: token0 } = useTotalCirculatingSupply(nftContract, 0);
const { data: token1 } = useTotalCirculatingSupply(nftContract, 1);
const { data: token2 } = useTotalCirculatingSupply(nftContract, 2);
const { data: token3 } = useTotalCirculatingSupply(nftContract, 3);

console.log({
  token0: token0?.toString(),
  token1: token1?.toString(),
  token2: token2?.toString(),
  token3: token3?.toString(),
});
```

## Changes made

- [ ] Public API changes: list the public API changes made if any
- [x] Internal API changes: explain the internal logic changes

## How to test

- [x] Automated tests: link to unit test file
- [ ] Manual tests: step by step instructions on how to test

### CODE REVIEW

File: packages/react-core/src/evm/utils/cache-keys.ts [hunk 1]
```diff
@@ -182,10 +182,12 @@ export const cacheKeys = {
           ),
         totalCirculatingSupply: (
           contractAddress: RequiredParam<ContractAddress>,
+          tokenId?: RequiredParam<BigNumberish>,
         ) =>
           createContractCacheKey(contractAddress, [
             "query",
             "totalCirculatingSupply",
+            tokenId ?? "erc721",
```
Comment (jnsdls) [inline packages/react-core/src/evm/utils/cache-keys.ts:190]:
```suggestion
            tokenId ?? 0,
```

tokenId will be a BigNumberIsh so I would simply go with `0` here for erc721 case (erc721 can only ever be one token so the "tokenId" would be `0` if it had one, seems reasonable)

also just a note for additional learning, no need to make this change here:
- the _correct_ hierarchy for the query key (cache key) here would be `contract -> query -> tokenId -> totalCirculatingSupply
- in this case it's OK to leave it as is as we actually never invalidate cache for  individual tokens anywhere today, but in general the principle should be "least significant -> "most significant"

**Review by jnsdls — APPROVED** [review]
small notes, overall looks good and good find!

File: packages/react-core/src/evm/utils/cache-keys.ts [hunk 2]
```diff
@@ -182,10 +182,12 @@ export const cacheKeys = {
           ),
         totalCirculatingSupply: (
           contractAddress: RequiredParam<ContractAddress>,
+          tokenId?: RequiredParam<BigNumberish>,
```
[the remark this answers is not in the document]
Comment (kien-ngo) [reply packages/react-core/src/evm/utils/cache-keys.ts:185]: ty ser

File: packages/react-core/src/evm/utils/cache-keys.ts (hunk 1 above)
Comment (kien-ngo) [reply packages/react-core/src/evm/utils/cache-keys.ts:190]: 🙏 🙏

**Pull request merged**